services/search/src/search/routes/filter.py

```diff
@@ -138,20 +138,21 @@ async def filter_endpoint(request: Request) -> Response:
                         hf_token=hf_token,
                     )
                 with StepProfiler(method="filter_endpoint", step="get features"):
-                    try:
-                        features = Features.from_dict(
-                            {
-                                name: feature
-                                for name, feature in duckdb_index_cache_entry["content"]["features"].items()
-                                if name != ROW_IDX_COLUMN
-                            }
-                        )
-                    except (KeyError, AttributeError):
-                        raise RuntimeError("The indexing process did not store the features.")
+                    if "features" in duckdb_index_cache_entry["content"] and isinstance(
+                        duckdb_index_cache_entry["content"]["features"], dict
+                    ):
+                        features = Features.from_dict(duckdb_index_cache_entry["content"]["features"])
```

Comment:
Note that you are now also including the `ROW_IDX_COLUMN`, contrary to what it was done before...

Reply:
Yes noted ! I added a test for create_response to take this into account

Reply:
IMO ROW_IDX_COLUMN is not part of the dataset features (we just added that column in the DuckDB table for convenience).

Reply:
The split-duckdb-index job does store ROW_IDX_COLUMN in the features

https://github.com/huggingface/datasets-server/blob/fc3e26e23d6d2aaa6dfa2f472a0276181370183d/services/worker/src/worker/job_runners/split/duckdb_index.py#L284-L285

Here I reuse the same features but since `use_row_idx_column=True` the ROW_IDX_COLUMN is not returned in `create_response`

Reply:
If the "features" field exists in the split-duckdb-index cache content, it contains ROW_IDX_COLUMN

If it doesn't exist, we do `Features.from_arrow_schema(pa_table.schema),` which also contains ROW_IDX_COLUMN

In any case `to_rows_list` takes care of removing the column from the row in the end

https://github.com/huggingface/datasets-server/blob/fc3e26e23d6d2aaa6dfa2f472a0276181370183d/libs/libapi/src/libapi/utils.py#L213